website_product_document_download_counter/models/product_document.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2025 Cetmix OÜ
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/LGPL).
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductDocument(models.Model):
+    _inherit = "product.document"
+
+    download_count = fields.Integer(
+        default=0,
+        readonly=True,
+        copy=False,
+        help="Number of times this document has been downloaded",
+    )
+
+    count_downloads = fields.Boolean(
+        copy=False,
+        help="Enable download counting for this document when published on website",
+    )
+
+    def increment_download_count(self):
+        """
+        Increment the download count for a specific product document.
+        Caller must ensure appropriate permissions (e.g. via sudo()).
+        """
+        self.ensure_one()
+
```

Comment:
It's better to not add empty lines inside a method, as having code syntax highlighting, they don't add value, but to make more scroll reading code.